File: tests/php/Search/DijkstrasTest.php
<?php

namespace GulDmitry\Algorithms\Tests\Search;

use GulDmitry\Algorithms\Search\Dijkstras;
use PHPUnit\Framework\TestCase;

class DijkstrasTest extends TestCase
{
    public function testSearch()
    {
        $graph = [];
        $graph['start'] = [];
        $graph['start']['a'] = 6;
        $graph['start']['b'] = 2;
        $graph['a'] = [];
        $graph['a']['fin'] = 1;
        $graph['b'] = [];
        $graph['b']['a'] = 3;
        $graph['b']['fin'] = 5;
        $graph['fin'] = [];

        $costs = [];
        $costs['a'] = 6;
        $costs['b'] = 2;
        $costs['fin'] = INF;

        $parents = [];
        $parents['a'] = 'start';
        $parents['b'] = 'start';
        $parents['fin'] = null;

        $expectedPath = [
            'fin' => 'a',
            'a' => 'b',
            'b' => 'start',
        ];

        $dijkstras = new Dijkstras();
        $actualPath = $dijkstras->search($graph, $costs, $parents);

        asort($expectedPath);
        asort($actualPath);
        $this->assertEquals($expectedPath, $actualPath);
    }
}

File: tests/go/Search/dijkstras_test.go
package Search

import "testing"
import search "github.com/guldmitry/algorithms/Search"
import (
	"math"
	"reflect"
)

func TestDijkstras(t *testing.T) {
	var graph = map[string]map[string]float64{
		"start": {
			"a": 6,
			"b": 2,
		},
		"a": {
			"fin": 1,
		},
		"b": {
			"a":   3,
			"fin": 5,
		},
		"fin": {},
	}
	var costs = map[string]float64{
		"a":   6,
		"b":   2,
		"fin": math.Inf(0),
	}
	var parents = map[string]string{
		"a":   "start",
		"b":   "start",
		"fin": "",
	}
	var expectedPath = map[string]string{
		"fin": "a",
		"a":   "b",
		"b":   "start",
	}

	actualPath := search.Dijkstras(graph, costs, parents)

	if !reflect.DeepEqual(expectedPath, actualPath) {
		t.Error("Invalid path has been built.")
	}
}
